components/sidebar_right/sidebar_right.tsx

```diff
@@ -244,33 +257,36 @@ export default class SidebarRight extends React.PureComponent<Props, State> {
         }
         const channelDisplayName = rhsChannel ? rhsChannel.display_name : '';
         const containerClassName = classNames('sidebar--right', {
-            'sidebar--right--expanded': isSidebarRightExpanded,
-            'move--left': isOpen,
-            hidden: !isOpen,
+            'sidebar--right--expanded expanded': isSidebarRightExpanded,
+            'move--left is-open': isOpen,
         });
 
         return (
-            <div
-                className={containerClassName}
-                id='sidebar-right'
-                role='complementary'
-                ref={this.sidebarRight}
-            >
-                <div
-                    onClick={this.onShrink}
-                    className='sidebar--right__bg'
-                />
-                <div className='sidebar-right-container'>
-                    <Search
-                        isSideBarRight={true}
-                        isSideBarRightOpen={this.state.isOpened}
-                        getFocus={this.getSearchBarFocus}
-                        channelDisplayName={channelDisplayName}
-                    >
-                        {isOpen && content}
-                    </Search>
-                </div>
-            </div>
+            <>
+                {isOpen && (
+                    <>
+                        <div className={'sidebar--right sidebar--right--width-holder'}/>
+                        <div
+                            className={containerClassName}
+                            id='sidebar-right'
+                            role='complementary'
+                            ref={this.sidebarRight}
+                        >
+                            <div className='sidebar-right-container'>
+                                <Search
+                                    isSideBarRight={true}
+                                    isSideBarRightOpen={true}
+                                    getFocus={this.getSearchBarFocus}
+                                    channelDisplayName={channelDisplayName}
+                                >
+                                    {content}
+                                </Search>
+                            </div>
+                        </div>
+                    </>
+                )}
+                <AppBar/>
```

Comment:
No, it could be a sibling and work just fine. 0/5 on that

`AppBar` has its own `return null` checks. 3/5 I think it's good that those checks are co-located with the main implementation, rather than add conditionals and more logic into root.

Reply:
Yeah, I agree with not adding it to the root. I do want to figure out a good pattern for conditionally mounting objects with as little overhead as possible while the component isn't visible, but that's outside the scope for this. There's a similar issue with menus and modals where we end up calculating a bunch of state for them and just rendering null most of the time since they aren't visible, and it'd be nice to tackle that more broadly

Reply:
Yeah there's definitely a balance. (Hoisted rendering conditionals in the parent vs in the child with return null.) Would be nice to have a standard/recommended pattern.